Update Register 5 to be configurable

rysiulg opened the pull request:
I hope i did as your code directive specified, removed spaces etc ;)

Mosibi commented on heatpump.yaml:
```diff
@@ -2233,33 +2174,276 @@ switch:
+          }
+  # Register: 5 -> Bit: 7
+  - platform: template
+    name: "Function Setting Silent Mode Level" #silent or very silent
```
Please remove the comment

Mosibi commented on heatpump.yaml:
```diff
@@ -2233,33 +2174,276 @@ switch:
+    name: "Function Setting Silent Mode Level" #silent or very silent
+    id: "${devicename}_function_setting_silent_mode_level"
+    icon: mdi:eye
+    #address: 5
```
Not needed anymore, can be removed, also the next line

Mosibi commented on heatpump.yaml:
```diff
@@ -2233,33 +2174,276 @@ switch:
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 5, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
+  # Register: 5 -> Bit 9 at binary sensor not used
```
Can you change this line to "Register: 5 -> Bit: 9 Is present in this config as a 'binary_sensor"

Mosibi commented on heatpump.yaml:
```diff
@@ -2233,33 +2174,276 @@ switch:
+  - platform: template
+    name: "Function Setting Holiday Home"
+    id: "${devicename}_function_setting_holiday_home"
+    disabled_by_default: true
```
I rather expose everything by default. It's up to the user if they use it in a dashboard or not. Please remove

Mosibi commented on heatpump.yaml:
```diff
@@ -2233,33 +2174,276 @@ switch:
+  - platform: template
+    name: "Function Setting ECO Mode"
+    id: "${devicename}_function_setting_eco_mode"
+    disabled_by_default: true
```
See other comment about this

rysiulg replied:
OK. One minute ;) -after commit i send register 210 next 211 ;)

Mosibi commented:
> I hope i did as your code directive specified, removed spaces etc ;)

Very good job. I am impressed!

Mosibi commented:
Looks good! Friday I have time to test your code and will merge it if it runs without issues (which I expect)

rysiulg commented:
Yes it will ;) Before submit i recompile to be sure that there no errors;) -I use it now with Register 210 and 211 also ;) and make for you smaller code from parts of my working now ;)
can you tell if you want some other sensors like; 
1. COP -calculation of energy used to energy consumed -this tell you how heatpump is efficient eg. value 3.69 means that for 1kWh of energy we have 3.69 Heat energy (similar for climate -there is parameter SOP for make cold)
2. I've added in my config also value which indicate when DHW starts to heat -to which temp T5 drops when heat pump start heat DHW -calculation T5s - delta T5s
3. Do you want uptime displayed in human readable format? days months hours days -this as second uptime -for homeassistant is standard version better to compare
4. I've run on ESP32 with led -so i use built in led to make status to wifi connected -blink at ~slow, connected to Homeassistant blink ~1,2s and not connected Homeassistant -other blink and no wifi connect no led -do you want that implementation?
5. I don't ask about automation -you write some time ago about that you don't want it ;) -i change to heat from october to april and then to cool my buffer  to cool by floor and recuperator ;), i decided to increase delta T5s for night (i have PV installation of 12kW so I wait for it for day). Also I've add to reanable Turn on Power in Register 0 for specified time eg. 3 hours after manually disable it, This is usefull also for that when you change some parameters like: Register 210, 211 delta temps and others can disable Power on all in register 0 ;) 

If it posiible give me answer for 1-5 Yes/No -so for yes at end i send you puul request for these single solutions ;) -this is good to get from one place goog code ;) -2 years ago i don't found anything that i write from scratch my implementation in arduino C++ + html + css + java -to be able to configure also by www ;) -now I moved to ESPHome ;) -i'm tired to get solution to wifi/ethernet reconnection etc. ;)
Writing this pull requests gives me to refresh code, catch a lot of errors -especially as advice to move esp-idf makes work ;)

rysiulg commented:
I've wondering -for your esp controller you get power from heatpump or external supply?

Mosibi commented:
1: yes
2: sounds nice, can you explain where you use this for?
3: I doubt about this one, with the current one it is very ease to spot when the controller has rebooted (in a graph). This one is meant to help debug issues instead of nicely showing the uptime. For now let's keep it out.
4. To keep this config generic, no let's not add that. For our own controller (which is sold on https://shop.svenar.nl), we do exactly the same. But for that config we use "patch" to inject it in the final config that is uploaded on those controllers.
5: No, those should be placed or done with home assistent imho.

Mosibi commented:
> I've wondering -for your esp controller you get power from heatpump or external supply?

The power comes from the heatpump self, although we've also have delivered a version with usb-c power for a special use case.

> I know electronic -i made own controller for control tyemperature to styart pumps for heating ;) 
> 
> Informatic knowledge as hobbist -also some code analysis for reverse engeeniering ;)
> 
> In work as service man (copiers, Wideformat ploters and scanners) in my own company for over 10 years;)

I already assumed that you had quite some experience in electronic and C(++) coding. Very cool!

gisbertg commented:
> I've wondering -for your esp controller you get power from heatpump or external supply? I know electronic -i made own controller for control tyemperature to styart pumps for heating ;) Informatic knowledge as hobbist -also some code analysis for reverse engeeniering ;) In work as service man (copiers, Wideformat ploters and scanners) in my own company for over 10 years;)

What does your controller look like? Maybe like mine? I have also connected the ESP and the Modbus converter directly in the controller.
[
](https://community.home-assistant.io/t/midea-branded-ac-s-with-esphome-no-cloud/265236/1161?u=gisbertg)

rysiulg commented:
> 1: yes 2: sounds nice, can you explain where you use this for? 3: I doubt about this one, with the current one it is very ease to spot when the controller has rebooted (in a graph). This one is meant to help debug issues instead of nicely showing the uptime. For now let's keep it out. 4. To keep this config generic, no let's not add that. For our own controller (which is sold on https://shop.svenar.nl), we do exactly the same. But for that config we use "patch" to inject it in the final config that is uploaded on those controllers. 5: No, those should be placed or done with home assistent imho.

in pnmt. 2. I've added this sensor instead of delta to see on dashboard when dhw start heating :)

Mosibi commented:
> in pnmt. 2. I've added this sensor instead of delta to see on dashboard when dhw start heating :)

Let’s make a separate PR for this, then I can look at it. An extra bit of info is welcome, it’s up to the user if they want to see/use it or not.

Mosibi commented on heatpump.yaml:
```diff
@@ -1202,135 +1220,59 @@ binary_sensor:
-    register_type: holding
-    address: 0x5
-    bitmask: 0x800
+    #disabled_by_default: true
```
Can you remove this one also.

Mosibi commented:
Tested and looks good!. Thanks for this one

Mosibi commented:
Fixes #25
